src/help/guides/extension-development/function-definitions.md

```diff
@@ -127,6 +159,76 @@ async getChapterDetails(mangaId: string, chapterId: string): Promise<ChapterDeta
 }
 ```
 
+## searchRequest
+
+### Method signature
+
+`async searchRequest(query: SearchRequest, metadata: any): Promise<PagedResults>`
+
+### Parameters
+
+| Parameter | Type | Description|
+|-----------|------|------------|
+| `query` | [SearchRequest](model-reference.md#searchrequest) | The object representing the search. View the fields of the [SearchRequest](model-reference.md#searchrequest) object to find out what a search can contain. |
+| `metadata` | any | Metadata carried through function calls. View the [metadata page](metadata.md) to learn more about how the parameter works. |
+
+### Returns
+
+A [PagedResults](model-reference#pagedresults) object with the results of the search for the current page, and metadata for the next page, if it exists.
+
+### Example Implementation
+
+```ts
+async searchRequest(query:SearchRequest, metadata:any): Promise<PagedResults> {
+	let page;
+
+	// Get the current page the function is on
+	if (typeof metadata === "object" && metadata.page) {
+		page = metadata.page;
+	} else { // If there is no current page, this must mean this is the first page.
+		page = 1;
+	}
+	const request = createRequest({
+		url: `https://mysite.com/search/?query=${query.title}&page=${page}`,
+		method: GET,
+		...
+	});
+	/*
+	Execute the request, and retrieve the HTML page
+	this.requestManager is provided to you by the parent class. Always use this to make requests.
+	The second parameter is the number of retries the app is allowed to make, if the request fails the first time.
+	*/
+	const data = await this.requestManager.schedule(request, 1)
+
+	// Prepare to parse the page using CheerioJS (Class object included by parent class)
```

Comment:
```suggestion
	// Prepare to parse the page using CheerioJS (The cheerio object is provided by the base class this extends)
```
I think this was a copy and paste from my original draft, lets do a slight update